src/components/ChatEntry.js

```diff
@@ -1,22 +1,56 @@
-import React from 'react';
-import './ChatEntry.css';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import './ChatEntry.css';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+  const [isLiked, setIsLiked] = useState(false);
+
+  const updateChatLog = () => {
+    props.onUpdateMessage({
+      id: props.id, 
+      sender: props.sender,
+      body: props.body,
+      timeStamp: props.timeStamp,
+      liked: !isLiked,
+    });
+
+    // Change the state of isLiked
+    setIsLiked(!isLiked);
+  };
+  
+  // Need to change heart emoji from white to red if message is liked
+  const heart = isLiked ? '❤️' : '🤍';
+
+  // Need to assign class 'local' if the sender is Vladimir and 'remote' if it's Estragon
+  const messageSide = props.sender === 'Vladimir' ? 'local' : 'remote';
+
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
-      <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+    <div className={`chat-entry ${messageSide}`}>
+      <h2 className="entry-name">
+        { props.sender }
+      </h2>
+      <section className='entry-bubble'>
+        <p>{ props.body }</p>
+        <p className="entry-time">
+          <TimeStamp time={ props.timeStamp }/>
+        </p>
+        <button 
+          onClick={ updateChatLog }
```

Comment:
Minor:

This line could also be `onClick=updateChatLog`. What you have also works, and it's nothing worth fretting over, but just wanted to share a "more direct syntax."

In your original line, `onClick={ updateChatLog }` is saying "the value of `onClick` should be the Javascript expression inside the `{}`, which evaluates as `updateChatLog`

In the line `onClick=updateChatLog`, it's saying that the value of `onClick` is the value of `updateChatLog`

Extremely minor, but maybe it's interesting!